cmd/agent/agent_create.go

```diff
@@ -0,0 +1,535 @@
+			Name:      defaultSecretName,
+			Namespace: o.Namespace,
+		},
+		Type: "string",
+		StringData: map[string]string{
+			"client-id":     o.credentials.ClientId,
+			"client-secret": o.credentials.ClientSecret,
+		},
+	}
+}
+
+// apply apply a configuration to a resource
+func (o *AgentOptions) apply(namespace, resourceFile string) error {
+	mapper, err := o.kubernetesFactory.ToRESTMapper()
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := o.kubernetesFactory.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	openAPISchema, _ := o.kubernetesFactory.OpenAPISchema()
+
+	printFlags := genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme)
+
+	// allow for a success message operation to be specified at print time
+	noopPrinter := func(operation string) (printers.ResourcePrinter, error) {
+		return printFlags.ToPrinter()
+	}
+
+	applyOptions := &apply.ApplyOptions{
+		PrintFlags: printFlags,
+
+		DeleteOptions: &delete.DeleteOptions{
+			FilenameOptions: resource.FilenameOptions{
+				Filenames: []string{
+					resourceFile,
+				},
+			},
+		},
+		ServerSideApply:   true,
+		FieldManager:      apply.FieldManagerClientSideApply,
+		Recorder:          genericclioptions.NoopRecorder{},
+		Namespace:         namespace,
+		EnforceNamespace:  true,
+		Builder:           o.kubernetesFactory.NewBuilder(),
+		Mapper:            mapper,
+		DynamicClient:     dynamicClient,
+		OpenAPISchema:     openAPISchema,
+		ToPrinter:         noopPrinter,
+		IOStreams:         genericclioptions.NewTestIOStreamsDiscard(),
+		VisitedUids:       sets.NewString(),
+		VisitedNamespaces: sets.NewString(),
+	}
+	return applyOptions.Run()
+}
+
+// secretExist check if agent is already installed on the cluster
+func (o *AgentOptions) secretExist() (bool, error) {
+	secretExistOptions := metav1.ListOptions{}
+	secret, err := o.KubernetesClient.Secrets(o.Namespace).List(o.Context, secretExistOptions)
+	if err != nil {
+		return false, err
+	}
+	_, exists := lo.Find(secret.Items, func(n corev1.Secret) bool {
+		return defaultSecretName == n.Name
+	})
+	return exists, nil
+}
+
+func (o *AgentOptions) generateManifests() (string, error) {
+	// create temp file
+	f, err := os.CreateTemp("", "rna-*.yaml")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	// download manifests
+	resp, err := http.Get(manifestTemplateDownloadUrl)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// check response
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("there was an error trying to download manifests; %s", resp.Status)
+	}
+
+	// write the template into the temp file
+	_, err = io.Copy(f, resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	// open the file
+	templateContent, err := os.ReadFile(f.Name())
+	if err != nil {
+		return "", err
+	}
+
+	var cntxt any = map[string]any{
+		"NAMESPACE":               o.Namespace,
+		"RNA_IDENTIFIER":          o.Name,
+		"APPLICATION_ENVIRONMENT": "prod",
```

Comment:
I am not sure what the best way to detect the environment is in the CLI, probably a hidden parameter?

Reply:
[through the addr var, ](https://github.com/armory-io/armory-cli/blob/bf3ee1f011e10cfb39c3fe5424ca17e88c22f67f/pkg/config/configuration.go#L48-L63)

Reply:
https://github.com/armory-io/armory-cli/blob/bf3ee1f011e10cfb39c3fe5424ca17e88c22f67f/pkg/config/configuration.go#L178-L209